rust/src/lib/nm/settings/bond.rs

```diff
@@ -232,8 +232,22 @@ pub(crate) fn gen_nm_bond_port_setting(
         return;
     };
 
+    // NetworkManager 1.34- does not support bond_port yet, we skip creating
+    // this NmSettingBondPort on default value unless it pre-exist.
+    if (bond_port_conf.priority.is_none() || bond_port_conf.priority == Some(0))
+        && (bond_port_conf.queue_id.is_none()
+            || bond_port_conf.queue_id == Some(0))
+        && nm_conn.bond_port.is_none()
+    {
+        return;
+    }
+
+    // NetworkManager 1.42.8- does not support priority, hence we do not touch
+    // it unless non-default defined or pre-exist.
     if let Some(v) = bond_port_conf.priority {
-        nm_set.priority = Some(v);
+        if nm_set.priority.is_some() || v != 0 {
```

Comment:
why not add to queue id ?  
  ```
  if nm_set.queue_id.is_some() || v != 0 {
  ```

Reply:
And I will test patch in vm

Reply:
The queue id is supported when bond_port feature been introduced into NM,
 * When NM does not support bond-port, it never exists in NM connection. So first check already return. If user defined non-zero value, they get InvalidArgument error.
 * When NM support bond-port, we can always set queue-id there.

Hence no need to do the same for queue_id.